RealexPayments/HPP/Controller/Apm/Result.php

```diff
@@ -0,0 +1,226 @@
+     */
+    private $_paymentManagement;
+
+    /**
+     * @var ResultFactory
+     */
+    private $_resultFactory;
+
+    /**
+     * Result constructor.
+     *
+     * @param Context                          $context
+     * @param Data                             $helper
+     * @param OrderRepository                  $orderRepository
+     * @param Logger                           $logger
+     * @param RealexPaymentManagementInterface $paymentManagement
+     * @param ResultFactory                    $resultFactory
+     */
+    public function __construct(
+        Context $context,
+        Data $helper,
+        OrderRepository $orderRepository,
+        Logger $logger,
+        RealexPaymentManagementInterface $paymentManagement,
+        ResultFactory $resultFactory
+    )
+    {
+        $this->_helper            = $helper;
+        $this->_orderRepository   = $orderRepository;
+        $this->_url               = $context->getUrl();
+        $this->_logger            = $logger;
+        $this->_paymentManagement = $paymentManagement;
+        $this->_resultFactory     = $resultFactory;
+        parent::__construct($context);
+    }
+
+    /**
+     * this is a silent action, there should not be any session or assumption, treat it like CLI
+     */
+    public function execute()
+    {
+        $resultRaw = $this->_resultFactory->create(ResultFactory::TYPE_RAW);
+        $resultRaw->setHttpResponseCode(503);
+
+        try {
+            $response = $this->getRequest()->getParams();
+
+            if ($response) {
+                $result = $this->_handleResponse($response);
+
+                if ($result === false) {
+                    // if no runtime exception and logic failed, resent notifications are just gonna fail again
+                    $resultRaw->setHttpResponseCode(200);
+                }
+            }
+        } catch (\Exception $e) {
+            $this->_logger->critical($e->getMessage());
+
+            $resultRaw->setHttpResponseCode(503);
+        }
+
+        return $resultRaw;
+    }
+
+    /**
+     * @param array $response
+     *
+     * @return bool
+     * @throws InputException
+     * @throws NoSuchEntityException
+     * @throws \Exception
+     */
+    private function _handleResponse($response)
+    {
+        if (empty($response)) {
+            $this->_logger->critical(__('Async - Empty response received from gateway'));
+
+            return false;
+        }
+
+        $this->_helper->logDebug(__('Async - Gateway response:') . print_r($this->_helper->stripTrimFields($response), true));
+
+        // validate response
+        $authStatus = $this->_validateResponse($response);
+        if (!$authStatus) {
+            $this->_logger->critical(__('Async - Invalid response received from gateway.'));
+
+            return false;
+        }
+        //get the actual order id
+        [$incrementId, $orderTimestamp] = explode('_', $response['orderid']);
```

Comment:
Given that ORDER_ID is passed as `$realOrderId . '_' . $timestamp` in the initial payload to HPP, the value returned by GP is supposed to be the same, so, it's safe to assume that it will respect the pattern. However, it would break if a merchant's site would keep IDs containing "_", but that would break regular payments as well.